From 005a9a78a8409de51408097321d0369427fc814f Mon Sep 17 00:00:00 2001
From: Ann Thornton <Ann.Thornton@freescale.com>
Date: Fri, 13 Feb 2009 15:13:17 -0600
Subject: [PATCH] ENGR00107844-1: regulator: allow search by regulator name

Signed-off-by: Ann Thornton <Ann.Thornton@freescale.com>
---
 drivers/regulator/core.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index 02a7744..d2a2306 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -927,6 +927,11 @@ struct regulator *regulator_get(struct device *dev, const char *id)
 			goto found;
 		}
 	}
+	list_for_each_entry(rdev, &regulator_list, list) {
+		if (strcmp(rdev->desc->name, id) == 0) {
+			goto found;
+		}
+	}
 	printk(KERN_ERR "regulator: Unable to get requested regulator: %s\n",
 	       id);
 	mutex_unlock(&regulator_list_mutex);
-- 
1.5.4.4

